src/goalServices/goals.js

```diff
@@ -416,6 +455,24 @@ export function reduceObjectives(newObjectives, currentObjectives = []) {
   return objectivesToSort;
 }
 
+const reduceRelationThroughActivityReportObjectives = (
+  objective,
+  join,
+  relation,
+  exists = {},
+) => {
+  const existingRelation = exists[relation] || [];
+  return uniqBy([
```

Comment:
maybe a common on this one at first glance hard to puzzle out [join] and [relation].

Reply:
sure